flujo/validation/linters.py

```
@@ -1759,6 +1759,399 @@ def analyze(self, pipeline: Any) -> Iterable[ValidationFinding]:
+
+    Detects step references (steps['name']) inside loop bodies that may not work as expected
+    due to loop scoping. In loop bodies, only previous_step or context should be used.
+    """
+
+    def analyze(self, pipeline: Any) -> Iterable[ValidationFinding]:
+        """Check for step references inside loop/map bodies."""
+        out: list[ValidationFinding] = []
+        steps = getattr(pipeline, "steps", []) or []
+
+        def _check_loop_body_steps(
+            body_steps: list[Any], loop_name: str, loop_meta: dict[str, Any]
```

Ruff (0.13.3)

Unused function argument: `loop_meta`

(ARG001)